src/main/java/gregtech/api/recipes/FluidKey.java

```diff
@@ -22,13 +22,20 @@ public boolean equals(Object o) {
         if (this == o) return true;
         if (!(o instanceof FluidKey)) return false;
         FluidKey fluidKey = (FluidKey) o;
-        return Objects.equals(fluid, fluidKey.fluid) &&
-                Objects.equals(tag, fluidKey.tag);
+        if (!Objects.equals(fluid, fluidKey.fluid) )
+            return false;
+        if (tag == null && fluidKey.tag != null) return false;
+        else return tag == null || tag.equals(fluidKey.tag);
```

Comment:
Shouldn't this check `(tag == null && fluidKey.tag == null) || tag.equals(fluidKey.tag)`

Reply:
IntelliJ says luidKey.tag == null is always true